dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/query/operators/TokenUtils.java

```diff
@@ -27,97 +27,71 @@
  */
 package org.hisp.dhis.query.operators;
 
-import java.util.Arrays;
+import static java.util.Arrays.asList;
+
 import java.util.List;
 
 import org.hibernate.criterion.MatchMode;
-import org.hisp.dhis.query.Type;
 
 /**
  * @author Henning Håkonsen
  */
 public class TokenUtils
 {
+    private TokenUtils()
+    {
+        throw new UnsupportedOperationException( "util" );
+    }
+
     public static List<String> getTokens( String value )
     {
-        return Arrays.asList( value.replaceAll( "[^\\p{L}0-9]", " " ).split( "[\\s@&.?$+-]+" ) );
+        return asList( value.replaceAll( "[^\\p{L}0-9]", " " ).split( "[\\s@&.?$+-]+" ) );
     }
 
     public static StringBuilder createRegex( String value )
     {
         StringBuilder regex = new StringBuilder();
 
-        List<String> tokens = TokenUtils.getTokens( value );
+        List<String> tokens = getTokens( value );
 
-        if ( tokens == null || tokens.isEmpty() )
+        if ( tokens.isEmpty() )
         {
             return regex;
         }
 
-        TokenUtils.getTokens( value ).forEach( token -> regex.append( "(?=.*" ).append( token ).append( ")" ) );
-
+        for ( String token : getTokens( value ) )
+        {
+            regex.append( "(?=.*" ).append( token ).append( ")" );
+        }
         return regex;
     }
 
-    public static <T> boolean test( List<T> args, T testValue, String targetValue, boolean caseSensitive,
-        MatchMode matchMode )
+    public static <T> boolean test( T value, String searchTerm, boolean caseSensitive, MatchMode mode )
     {
-        if ( args.isEmpty() || testValue == null )
```

Comment:
`args` cannot be empty at this point since otherwise the construction of the operator would have failed. It was therefore removed as a parameter from this method.